File: base/src/main/groovy/com/mercadolibre/android/gradle/base/modules/SettingsModule.java
package com.mercadolibre.android.gradle.base.modules;

import org.gradle.api.initialization.Settings;

/**
 * Base interface of a settings module
 * <p>
 * Created by mafunes on 10/08/20.
 */
interface SettingsModule {
    void configure(Settings settings);
}

File: baseplugin/src/main/java/com/mercadolibre/android/gradle/baseplugin/core/domain/interfaces/SettingsModule.kt
package com.mercadolibre.android.gradle.baseplugin.core.domain.interfaces

import org.gradle.api.initialization.Settings

/**
 * This interface is responsible for standardizing the modules so that they configurer settings correctly.
 */
interface SettingsModule {
    /**
     * This method is responsible for requesting all the modules that can configure a settings.
     */
    fun configure(settings: Settings)
}
